docs/architecture/adr-001-initial-QGB-p2p-network.md

```diff
@@ -0,0 +1,195 @@
+# ADR 001: Initial QGB P2P network
+
+## Context
+
+The QGB V2 will rely on a P2P network to exchange attestations signatures. This comes after deciding that using the state will result in state bloat, and using block data will always have an opportunity cost for validators. Thus, we will rely on a P2P network for exchanging attestations' signatures between orchestrators and relayers.
+
+An attestation signature can be identified by the `(nonce,orchestrator,type)` tuple:
+
+- `nonce`: nonce of the attestation
+- `orchestrator`: orchestrator address
+- `type`: data commitment confirm or valset confirm
+
+The issue with using a P2P network is that there is no way of knowing the CID of a signature beforehand. So, relayers will not be able to get the confirms directly from the network.
```

Comment:
there's no way of knowing the _hash_ before hand, but if we define a custom CID format (similar to the plugin), then we can know the CID ahead of time!

also, mind clarifying why we can't know the hash ahead of time to readers of this document?